thingsboard_gateway/connectors/ftp/backward_compatibility_adapter.py

```diff
@@ -0,0 +1,93 @@
+from copy import deepcopy
+import json
+
+
+class FTPBackwardCompatibilityAdapter:
+
+    def __init__(self, config: dict) -> None:
+        self._config = deepcopy(config)
+        self._converted_config = {
+            "parameters": {
+            },
+            "paths": [],
+            "requestsMapping": {
+                "attributeUpdates": [],
+                "serverSideRpc": []
+            }
+        }
+
+    def convert(self) -> dict:
+
+        self._converted_config['parameters']['host'] = self._config.get('host')
+        self._converted_config['parameters']['port'] = self._config.get('port')
+        self._converted_config["parameters"]["TLSSupport"] = self._config.get("TLSSupport", False)
+        security = self._config.get('security', {})
+        self._converted_config['parameters']['security'] = security
+        self._converted_config["parameters"]["security"]["type"] = security.get("type")
```

Comment:
username and password may be only if type is basic, they are not necessary for anonymous.